search-backend-remoterepository/src/main/java/org/apache/maven/search/backend/remoterepository/internal/RemoteRepositorySearchBackendImpl.java

```diff
@@ -0,0 +1,388 @@
+                                        ext = name.substring(name.lastIndexOf('.') + 1);
+                                    } else {
+                                        ext = name.substring(name.lastIndexOf('.') + 1) + "." + ext;
+                                    }
+                                    name = name.substring(0, name.lastIndexOf('.'));
+                                }
+                                classifier = name.isEmpty() ? null : name;
+                            } else {
+                                ext = name;
+                            }
+                            page.add(create(
+                                    context.getGroupId(),
+                                    context.getArtifactId(),
+                                    context.getVersion(),
+                                    classifier,
+                                    ext));
+                        }
+                    }
+                }
+            }
+        }
+        return page.size();
+    }
+
+    /**
+     * Creates a {@link Record} instance using passed in field values. All field values except
+     * {@code groupId} are optional (nullable).
+     */
+    protected Record create(
+            String groupId, String artifactId, String version, String classifier, String fileExtension) {
+        HashMap<Field, Object> result = new HashMap<>();
+
+        mayPut(result, MAVEN.GROUP_ID, groupId);
+        mayPut(result, MAVEN.ARTIFACT_ID, artifactId);
+        mayPut(result, MAVEN.VERSION, version);
+        mayPut(result, MAVEN.CLASSIFIER, classifier);
+        mayPut(result, MAVEN.FILE_EXTENSION, fileExtension);
+        return new Record(getBackendId(), getRepositoryId(), null, null, result);
+    }
+
+    private static String readChecksum(InputStream inputStream) throws IOException {
+        String checksum = "";
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8), 512)) {
+            while (true) {
+                String line = br.readLine();
+                if (line == null) {
+                    break;
+                }
+                line = line.trim();
+                if (line.length() > 0) {
+                    checksum = line;
+                    break;
+                }
+            }
+        }
+
+        if (checksum.matches(".+= [0-9A-Fa-f]+")) {
+            int lastSpacePos = checksum.lastIndexOf(' ');
+            checksum = checksum.substring(lastSpacePos + 1);
+        } else {
+            int spacePos = checksum.indexOf(' ');
+
+            if (spacePos != -1) {
+                checksum = checksum.substring(0, spacePos);
+            }
+        }
+
+        return checksum;
+    }
+
+    private static void mayPut(Map<Field, Object> result, Field fieldName, Object value) {
+        if (value == null) {
+            return;
+        }
+        if (value instanceof String && ((String) value).trim().isEmpty()) {
```

Comment:
super nitpick:
`.trim().isEmpty()` -> `.isBlank()`

I only noticed since I have a small collection of refactoring rules here: https://github.com/mbien/jackpot-inspections which fired on this.